### DISCUSSION TITLE
Generate typegraphql from typescript.

### DISCUSSION THREAD

**Discussion by valerii15298** [MichalLytek/type-graphql#1259] [Q&A] [answered]
Is there a way to generate Typegraphql from typescript types?

Basically i have library which has types and i need to build graphql api on top of that. And creating objectTypes manually is not very comfortable... Since there are lot of types. So having ability to generate typegraphql from typescript could be really helpful.
Is there some tools to achieve that?
Or if not then what could be alternative to such problem?
Writing manually is gonna be not very ideal, and if types will change I will need to manually update typegraphql every time...

**Comment by MichalLytek** [comment, marked as answer]
I think you should write your own plugin for GraphQL Code Generator
https://www.graphql-code-generator.com/plugins/typescript-type-graphql

**Reply by valerii15298** [reply]
Yep:) Though all plugins in graphql-code-generator only generate code from graphql. But not in other direction. And this one you mentioned also generates from graphql. As I see, for now there is no good plugins even to [generate](https://stackoverflow.com/questions/64637975/generate-graphql-schema-from-typescript) graphql from typescript. Anyway I solved my problem just by hand for now.

**Reply by valerii15298** [reply]
@MichalLytek I think I am ready start at least investigation on it.
Are there any recommendations of where to start from? I did not write any codegen plugins, so will start looking into existing plugins to understand how it works.

**Reply by MichalLytek** [reply]
I haven't either. Ask on their repo.
